src/home-assistant.html

```diff
@@ -50,70 +50,70 @@
   }
 };
 
-Polymer({
-  is: 'home-assistant',
-
-  hostAttributes: {
-    icons: null,
-  },
-
-  properties: {
-    connectionPromise: {
-      type: Object,
-      value: window.hassConnection || null,
-      observer: 'handleConnectionPromise',
-    },
-    connection: {
-      type: Object,
-      value: null,
-      observer: 'connectionChanged',
-    },
-    hass: {
-      type: Object,
-      value: null,
-    },
-    icons: {
-      type: String,
-    },
-    iconsLoaded: {
-      type: Boolean,
-      value: false,
-    },
-    showMain: {
-      type: Boolean,
-      computed: 'computeShowMain(hass, iconsLoaded)',
-    },
-  },
-
-  listeners: {
-    settheme: 'setTheme',
-  },
-
-  computeShowMain: function (hass, iconsLoaded) {
+class HomeAssistant extends Polymer.Element {
+
+  static get is() { return 'home-assistant'; }
+
+  static get properties() {
+    return {
+      connectionPromise: {
+        type: Object,
+        value: window.hassConnection || null,
+        observer: 'handleConnectionPromise',
+      },
+      connection: {
+        type: Object,
+        value: null,
+        observer: 'connectionChanged',
+      },
+      hass: {
+        type: Object,
+        value: null,
+      },
+      icons: {
+        type: String,
+      },
+      iconsLoaded: {
+        type: Boolean,
+        value: false,
+      },
+      showMain: {
+        type: Boolean,
+        computed: 'computeShowMain(hass, iconsLoaded)',
+      },
+    };
+  }
+
+  ready() {
+    super.ready();
+    this.addEventListener('settheme', e => this.setTheme(e));
+    this.loadIcons();
+  }
+
+  computeShowMain(hass, iconsLoaded) {
```

Comment:
The linter is still complaining about these. Because Polymer only calls instance methods, not static methods, for computed properties they can't be switched to static. The resolution might just be to disable that linting rule. Open to ideas here.

Reply:
Yes, let's disable that rule.

I initially based the linting rules on airbnb but that one is not very Polymer friendly.